src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt

```diff
@@ -209,22 +211,30 @@ class DtlsStack(
         onOutgoingProtocolData(listOf(packet))
     }
 
+    override fun getNodeStats(): NodeStatsBlock = NodeStatsBlock("DtlsStack").apply {
+        addBlock(NodeStatsBlock("localFingerprint").apply {
+            addString(certificateInfo.localFingerprintHashFunction, certificateInfo.localFingerprint)
+        })
+        addBlock(NodeStatsBlock("remoteFingerprints").apply {
+            remoteFingerprints.forEach { (hash, fp) -> addString(hash, fp) }
+        })
+        addString("role", (role?.javaClass ?: "null").toString())
+    }
+
     companion object {
         /**
          * Because generating the certificateInfo can be expensive, we generate a single
          * one to be used everywhere which expires in 24 hours (when we'll generate
          * another one).
          */
-        private var certificateInfo: CertificateInfo = DtlsUtils.generateCertificateInfo()
         private val syncRoot: Any = Any()
-        fun getCertificateInfo(): CertificateInfo {
-            synchronized(DtlsStack.syncRoot) {
+        private var certificateInfo: CertificateInfo = DtlsUtils.generateCertificateInfo()
+            get() = synchronized(syncRoot) {
                 val expirationPeriodMs = Duration.ofDays(1).toMillis()
-                if (DtlsStack.certificateInfo.creationTimestampMs + expirationPeriodMs < System.currentTimeMillis()) {
-                    DtlsStack.certificateInfo = DtlsUtils.generateCertificateInfo()
+                if (field.creationTimestampMs + expirationPeriodMs < System.currentTimeMillis()) {
+                    Thread { field = DtlsUtils.generateCertificateInfo() }.start()
```

Comment:
why would we want to do this asynchronously, we need to generate it before we return it if it's expired, no?

Reply:
It's not "expired" in the real sense, we've just decided that we want to refresh it every 24h. I return the existing one so we can proceed with immediately (instead of blocking the connection to an endpoint while waiting).

Reply:
we've got a race condition this way though

Reply:
yeah, we might end up generating a couple of certificates in vain

Reply:
i worry it is also error prone if someone decides to make multiple calls to certificateInfo to retrieve it, since we could get different results.

also, JMT currently does zero thread creation, which is a invariant i think would be nice to maintain.  do we know how costly this call is?

Reply:
"someone decides to make multiple calls to certificateInfo "
That's exactly the problem I'm solving. The static var is now private, and it is only accessed from DtlsStack when it initializes it's own val.

> also, JMT currently does zero thread creation, which is a invariant i think would be nice to maintain. do we know how costly this call is?

I'll add a note, because I don't want this to execute in the main thread (it will slow down conference creation), but I don't want to spend time on the necessary plumbing right now.